lib/core/uri_parser.js

```diff
@@ -680,11 +680,22 @@ function parseConnectionString(uri, options, callback) {
     return callback(new MongoParseError('No hostname or hostnames provided in connection string'));
   }
 
-  if (hosts.length !== 1 && parsedOptions.directConnection) {
-    // If the option is set to true, the driver MUST validate that there is exactly one host given in the host list in the URI, and fail client creation otherwise.
+  const directConnection = !!parsedOptions.directConnection;
```

Comment:
Curious: why use `!!` here?

Reply:
just to ensure its a boolean, it's not technically necessary